libmultilabel/nn/networks/labelwise_attention_networks.py

```diff
@@ -0,0 +1,139 @@
+        super(BiGRULWAN, self).__init__(embed_vecs, num_classes, dropout,
+                                        rnn_dim)
+
+    def _get_encoder(self, input_size, dropout):
+        assert self.rnn_dim % 2 == 0, """`rnn_dim` should be even."""
+        return GRUEncoder(input_size, self.rnn_dim // 2, self.rnn_layers,
+                          dropout)
+
+
+class BiLSTMLWAN(LabelwiseAttentionNetwork):
+    """BiLSTM Labelwise Attention Network
+
+    Args:
+        embed_vecs (FloatTensor): The pre-trained word vectors of shape (vocab_size, embed_dim).
+        num_classes (int): Total number of classes.
+        rnn_dim (int): The size of bidirectional hidden layers. The hidden size of the LSTM network
+            is set to rnn_dim//2. Defaults to 512.
+        rnn_layers (int): Number of recurrent layers. Defaults to 1.
+        dropout (float): The dropout rate of the word embedding. Defaults to 0.2.
+        activation (str): Activation function to be used. Defaults to 'tanh'.
+    """
+
+    def __init__(
+        self,
+        embed_vecs,
+        num_classes,
+        rnn_dim=512,
+        rnn_layers=1,
+        dropout=0.2,
+    ):
+        self.rnn_dim = rnn_dim
+        self.rnn_layers = rnn_layers
+        super(BiLSTMLWAN, self).__init__(embed_vecs, num_classes, dropout,
+                                         rnn_dim)
+
+    def _get_encoder(self, input_size, dropout):
+        assert self.rnn_dim % 2 == 0, """`rnn_dim` should be even."""
+        return LSTMEncoder(input_size, self.rnn_dim //
+                           2, self.rnn_layers, dropout)
+
+
+class CNNLWAN(LabelwiseAttentionNetwork):
+    """CNN Labelwise Attention Network
+
+    Args:
+        embed_vecs (FloatTensor): The pre-trained word vectors of shape (vocab_size, embed_dim).
+        num_classes (int): Total number of classes.
+        filter_sizes (list): Size of convolutional filters.
+        num_filter_per_size (int): Number of filters in convolutional layers in each size. Defaults to 50.
+        dropout (float): The dropout rate of the word embedding. Defaults to 0.2.
+        activation (str): Activation function to be used. Defaults to 'tanh'.
+    """
+
+    def __init__(
+        self,
+        embed_vecs,
+        num_classes,
+        filter_sizes=None,
+        num_filter_per_size=50,
+        dropout=0.2,
+        activation='tanh'
+    ):
+        self.filter_sizes = filter_sizes
+        self.num_filter_per_size = num_filter_per_size
+        self.activation = activation
+        hidden_dim = num_filter_per_size * len(filter_sizes)
+        super(CNNLWAN, self).__init__(embed_vecs, num_classes, dropout,
+                                      hidden_dim)
+
+    def _get_encoder(self, input_size, dropout):
+        return CNNEncoder(input_size, self.filter_sizes,
+                          self.num_filter_per_size, self.activation,
+                          channel_last=True)
+
+    def forward(self, input):
```

Comment:
cannot we just use forward of LabelwiseAttentionNetwork?

Reply:
Because the RNN encoders require lengths but CNN is not.